cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -398,29 +401,47 @@ var (
 )
 
 func isCoreDNSVersionSupported(client clientset.Interface) (bool, error) {
-	isValidVersion := true
-	coreDNSPodList, err := client.CoreV1().Pods(metav1.NamespaceSystem).List(
-		context.TODO(),
-		metav1.ListOptions{
-			LabelSelector: "k8s-app=kube-dns",
-		},
-	)
-	if err != nil {
-		return false, errors.Wrap(err, "unable to list CoreDNS pods")
+	var lastError error
+	var pods []v1.Pod
+	err := wait.PollImmediate(constants.APICallRetryInterval, 10 * time.Second, func() (bool, error) {
+		coreDNSPodList, err := client.CoreV1().Pods(metav1.NamespaceSystem).List(
+			context.TODO(),
+			metav1.ListOptions{
+				LabelSelector: "k8s-app=kube-dns",
+				FieldSelector: "status.phase=Running",
+			},
+		)
+		if err != nil {
+			lastError = err
+			return false, nil
+		}
+
+		if len(coreDNSPodList.Items) <= 0 {
+			lastError = errors.New("unable to list CoreDNS Running pods")
+			return false, nil
+		}
+		pods = coreDNSPodList.Items
+		return true, nil
+	})
+
+	if err != nil || len(pods) <= 0 {
+		return false, fmt.Errorf("%v: %v", err, lastError)
```

Comment:
`errors.Wrapf(lastError, "could not find any Running CoreDNS pods after %v", pollTimeout)`

and `pollTimeout` can be set to `10 * time.Second` above the poll.